Source/WTF/wtf/cocoa/WorkQueueCocoa.cpp

```diff
@@ -56,14 +56,11 @@ void WorkQueueBase::dispatch(Function<void()>&& function)
 
 void WorkQueueBase::dispatchWithQOS(Function<void()>&& function, QOS qos)
 {
-    dispatch_block_t blockWithQOS = dispatch_block_create_with_qos_class(DISPATCH_BLOCK_ENFORCE_QOS_CLASS, Thread::dispatchQOSClass(qos), 0, makeBlockPtr([function = WTFMove(function)] () mutable {
+    OSObjectPtr blockWithQOS = dispatch_block_create_with_qos_class(DISPATCH_BLOCK_ENFORCE_QOS_CLASS, Thread::dispatchQOSClass(qos), 0, makeBlockPtr([function = WTFMove(function)] () mutable {
```

Comment:
This leaks because `dispatch_block_create_with_qos_class()` has “create” in its name.

You need to use `adoptOSObject()` here.

Reply:
Perhaps emphasizing how fraught it is to reason about ARC in a mixed-language / mixed-mode project, I don't think this is a leak. ARC is enabled in this file, and therefore ARC "owns" the local +1 created by "create", and will release it at end of scope. Relatedly, `retainOSObject()`, called by the `OSObjectPtr` constructor, is a no-op.

Still, good to mirror our current coding idioms. I'll fix.

Reply:
LOL!  Yeah, it's going to be nearly impossible to review code without knowing whether each file is ARC or MRR.

(Even if you put the compiler check at the top of the file, it still means you need to click your way to the top of the file to check during reviews.)

Reply:
I suppose this is another argument for not using `RetainPtr<>` in ARC sources, though it still has the same issue of not knowing which files have ARC enabled when reviewing.